Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,318 @@
+    {
+        // hide cancel and back buttons when last page succeed
+        if (currentPage() == m_prefabMakerPage)
+        {
+            QWizard::button(QWizard::CancelButton)->hide();
+            QWizard::button(QWizard::BackButton)->hide();
+            QWizard::button(PrefabCreationButtonId)->hide();
+        }
+    }
+
+    void RobotImporterWidget::OpenUrdf()
+    {
+        m_urdfPath = AZStd::string(m_fileSelectPage->getFileName().toUtf8().constData());
+        if (!m_urdfPath.empty())
+        {
+            AZ_Printf("Wizard", "Loading URDF file : %s", m_urdfPath.c_str());
+            m_parsedUrdf = UrdfParser::ParseFromFile(m_urdfPath);
+            QString report;
+            const auto log = UrdfParser::GetUrdfParsingLog();
+            if (m_parsedUrdf)
+            {
+                report += "# " + tr("The URDF was parsed and opened successfully") + "\n";
+                // get rid of old prefab maker
+                m_prefabMaker.reset();
+                // let us skip this page
+                AZ_Printf("Wizard", "Wizard skips m_checkUrdfPage since there is no errors in URDF");
+                m_meshNames = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), true, true);
+            }
+            else
+            {
+                report += "# " + tr("The URDF was not opened") + "\n";
+                report += tr("URDF parser returned following errors:") + "\n\n";
+            }
+            if (!log.empty())
+            {
+                report += "`";
+                report += QString::fromUtf8(log.data(), int(log.size()));
+                report += "`";
+            }
+            m_checkUrdfPage->ReportURDFResult(report, m_parsedUrdf != nullptr);
+        }
+    }
+
+    void RobotImporterWidget::onCurrentIdChanged(int id)
+    {
+        AZ_Printf("Wizard", "Wizard at page %d", id);
+
+        if (currentPage() == m_assetPage)
+        {
+            FillAssetPage();
+        }
+        else if (currentPage() == m_prefabMakerPage)
+        {
+            FillPrefabMakerPage();
+        }
+    }
+
+    void RobotImporterWidget::FillAssetPage()
+    {
+        m_assetPage->ClearAssetsList();
+        if (m_parsedUrdf)
+        {
+            m_urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>(Utils::FindAssetsForUrdf(m_meshNames, m_urdfPath));
+            auto colliders_names = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), false, true);
+            auto visual_names = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), true, false);
+            for (const AZStd::string& mesh_path : m_meshNames)
+            {
+                const QString mesh_pathqs = QString::fromUtf8(mesh_path.data(), mesh_path.size());
+                const QString kNotFound = tr("not found");
+
+                if (m_urdfAssetsMapping->contains(mesh_path))
+                {
+                    QString type = kNotFound;
+                    QString source_path = kNotFound;
```

Comment:
The `source_path` variable should be lowerCamelCase: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names
```suggestion
                    QString sourcePath = kNotFound;
```